aten/src/ATen/core/ivalue.cpp

```diff
@@ -1,3 +1,7 @@
+#ifdef Q_SLOTS
+#undef slots
+#endif
+
```

Comment:
Why this change is necessary?

Reply:
slots is an alias for Q_SLOTS.This macro definition works globally, replacing all “slots” in the header file, including variable names.So, in this case, when the qt header file and the torch header file work together, they will be compiled together, and this slot will replace all the characters in the torch header file with all the names of the slots, which is very scary, and there will be a lot of compilation errors, syntax errors such as unclosed parentheses, etc

Reply:
I redefine this Marco "slots" at the end of the header.
In this case, the source file also needs to be undefined. Because he also contains the "slots" character.